### ISSUE TITLE
[TW-835] timestamp for status changes

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#870]
_Cory Donnelly on 2009-11-18T11:29:39Z says:_ 

 (was: All statuses should store an associated date value)

Currently in v1.8.3, completing a task causes the task to be moved to completed.data, status is set to "completed", and an end value (end:"1258403641" for example) is added.  This could be handled more generally.

For all statuses, when status is changed, an associated date value should be set (statusdate:"1258403641" perhaps?) This allows us to ask questions like,

"How long have I been waiting for person X to respond to this email?"
"When did I decide to retire this task?"
"When did this task (which was previously set to wait) wake up and become pending again?"

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2009-11-18T11:29:39Z
Modified: 2014-02-09T02:06:39Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2009-11-18T11:45:59Z says:_ 

 Interestingly, we already have this information in the undo log.  How would you like to see it presented?  Just on the "info" report?  Somewhere else?

**Comment by taskwarrior** [comment]
_David Patrick on 2009-11-18T14:48:29Z says:_ 

 Cory Donnelly wrote:
> For all statuses, when status is changed, an associated date value should be set (statusdate:"1258403641" perhaps?)
I think that with the combination of existing task and history data, and minor enhancements of existing reports, some of the questions you  pose should be answerable.

> This allows us to ask questions like,
> 
> "How long have I been waiting for person X to respond to this email?"
report.waiting could have an age field.

> "When did I decide to retire this task?"
report.retired or report.status would need to be created 
OR (and I like this better)
task history might learn to take a task ID argument, as in

 task history 10

to see the modification history of that one task.

> "When did this task (which was previously set to wait) wake up and become pending again?"

see: above
* ---

and I just discovered that this already works !

 task history proj:foo


djp

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2009-12-04T22:58:42Z says:_ 

 Changed title - "Report Enhancements" too generic.

**Comment by taskwarrior** [comment]
_Peter De Poorter on 2010-11-13T07:32:41Z says:_ 

 Why not creating a general log for logging of all activity ?
This log can be used as base for the asked reporting, auditing or can just be consulted by the user. I had the same idea when working with the stop, start journaling. I wanted to use it to register the time I spent to a task. After several stop, starts the annotation list becomes too long and I started thinking how this could be done more generally. Creating a log and writing reports on it was the solution direction I was going to.
